modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/220_failure_store_cluster_setting.yml

```diff
@@ -0,0 +1,243 @@
+setup:
+  - requires:
+      reason: "Data stream options was added in 8.18+"
+      test_runner_features: [ capabilities, allowed_warnings, contains ]
+      capabilities:
+        - method: POST
+          path: /{index}/_doc
+          capabilities: [ 'failure_store_status' ]
+        - method: POST
+          path: /_index_template/{template}
+          capabilities: [ 'failure_store_in_template' ]
```

Comment:
So, yeah, you're right, I was being a bit lazy and piggy-backing on the `failure_store_in_template` capability. You only added that last week, so I do think they have a good chance of appearing in the same version... But it's not certain. I've pushed a commit which adds a node feature for this... Let me know what you think.

I do have a couple of follow-up questions:

1. Should do `assumeTrue(clusterHasFeature(...))` in the Java REST test `DataStreamOptionsIT`? It would be easy enough, but we don't seem to have an equivalent check for the API capability right now, so I'm not sure.
2. I'm not quite sure _why_ we have this `requires` block in the YAML test. Do we ever run tests from `HEAD` against an older server or a server without all feature flags enabled? From my limited experimentation, it seems like the effect of the `requires` block would be to make the test fail-fast if we did, anyway.

Reply:
One of the differences between YAML tests and Java REST tests is that YAML tests are also run in bwc, meaning they run against older versions of elasticsearch. This is why we use these "requirements" to ensure that the tests are going to be run against versions that support the feature.

Also this is why, you do not need this in Java REST test.

Now going back to the node feature. I think we need to think as well, do we need this node feature in the production code? If not, we can rethink if it's useful to add it or not.

Reply:
But BWC is about running an old test suite against a new cluster, right?

Feature flags are enabled (and can't be disabled) in all snapshot builds: https://github.com/elastic/elasticsearch/blob/8134c79ce8a11b26ee26d88ad7975c91a127af98/server/src/main/java/org/elasticsearch/common/util/FeatureFlag.java#L72. So long as you're running the tests with a snapshot build rather than a release build, they should all be on. Unless I've misunderstood... These tests are certainly running locally for me!

Reply:
> Now going back to the node feature. I think we need to think as well, do we need this node feature in the production code? If not, we can rethink if it's useful to add it or not.

I don't really understand the node features framework well enough to answer this question. I'll defer to you, or we can go ask someone who knows this stuff better.

Reply:
(I can easily revert that commit if we don't need it.)

Reply:
> But BWC is about running an old test suite against a new cluster, right?

No, if I understand it correctly, we run the new test suite to previous versions, including mixed clusters.

> Feature flags are enabled (and can't be disabled) in all snapshot builds:

That is true, but we also run the test suites as release, so we explicitly enable the feature flags because we might end up with mixed versions.

> I don't really understand the node features framework well enough to answer this question. I'll defer to you, or we can go ask someone who knows this stuff better.

There are certain features that need to be supported by all nodes before they can take effect. For example, hypothetically, we might want to allow users to use the failure store, only when all nodes support the failure store. Otherwise, you might have a node adding failure store but the bulk request goes to a different node that doesn't know about the failure store. This can be ensure via a node feature. 

A node feature can be used also in yaml tests to see if the feature is supported.

So, I was wondering here, considering that:

- We are still behind a feature flag.
- I think, without being 100% sure, that the capabilities of these APIs will remain there and will be in the same version as this.

Maybe it depends if we are planning to introduce a general node feature for the failure store. @jbaiera & @dakrone . Any thoughts? 

Could we bundle this under a failure store node feature when we remove the feature flag?

Reply:
Yes, having slept on it, I think this:

> Could we bundle this under a failure store node feature when we remove the feature flag?

is the conclusion I'm leaning towards as well.

Reasoning:

1. We will not remove the feature flag until the implementation is complete.
2. Features are not, as I understand it, considered supported while they're behind a feature flag (the user would have to set the property or use a snapshot build, and figure out how to use it without documentation).
3. _Therefore_ we don't need to support the case of a mixed cluster in which the feature is enabled on some/all nodes and some/all nodes only have a partial implementation.
4. _Therefore_ there is no need for a node feature for the cluster setting bit itself, and I should revert the commit which added it.
5. _However_ we might want to support the case of a mixed cluster in which the feature is fully-implemented and enabled by default on some nodes and disabled on others (e.g. if we finish the implementation and remove the feature flag in 8.18, and the cluster has a mix of 8.17 and 8.18).
6. _Therefore_ we might need to use the node feature mechanism for the whole failure store feature.
7. _But_ that shouldn't be part of this PR — perhaps it should be part of the PR which removes the feature flag, just replacing checks on the feature flag with checks on the cluster feature (so it'll be enabled only if every node has the feature).

Reply:
As discussed in the meeting, I've removed the node feature and added a capability on the `PUT /_cluster/settings` API instead (with a comment mentioning that we plan to replace this when we remove the feature gate).